File: field-manager-dotnet-backend/Controllers/NutritionController.cs
using FieldManagerDotnetBackend.Models;
using FieldManagerDotnetBackend.Services;
using Microsoft.AspNetCore.Mvc;
using System;

namespace FieldManagerDotnetBackend.Controllers
{
    /// <summary>
    /// Provides nutrient recommendations and fertilizer suggestions for a
    /// field. This mimics the "Crop Nutrition" feature of xarvio FIELD
    /// MANAGER, where users can create field-zone specific application
    /// maps for nutrients.
    /// </summary>
    [ApiController]
    [Route("api/fields/{fieldId:guid}/nutrition")]
    public class NutritionController : ControllerBase
    {
        private readonly DemoDataService _dataService;

        public NutritionController(DemoDataService dataService)
        {
            _dataService = dataService;
        }

        [HttpGet]
        public ActionResult<NutritionRecommendation> GetNutrition(Guid fieldId)
        {
            var field = _dataService.GetField(fieldId);
            if (field is null || field.NutritionRecommendation is null)
            {
                return NotFound();
            }
            return Ok(field.NutritionRecommendation);
        }
    }
}

File: fieldmanager-spring-backend/src/main/java/com/xarvio/fieldmanager/controllers/NutritionController.java
package com.xarvio.fieldmanager.controllers;

import java.util.UUID;
import lombok.RequiredArgsConstructor;

import com.xarvio.fieldmanager.models.Field;
import com.xarvio.fieldmanager.models.NutritionRecommendation;
import com.xarvio.fieldmanager.services.DemoDataService;

import org.springframework.web.bind.annotation.RestController;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;

@RestController
@RequiredArgsConstructor
public class NutritionController {

    private final DemoDataService demoDataService;

    @GetMapping("api/fields/{fieldId}/nutrition")
    public ResponseEntity<NutritionRecommendation> getNutrition(@PathVariable UUID fieldId) {
        return demoDataService.getField(fieldId)
                .map(Field::nutrientRequirements)
                .map(ResponseEntity::ok)
                .orElse(ResponseEntity.notFound().build());
    }

}
